redis/src/cluster.rs

```diff
@@ -95,25 +96,23 @@ impl TlsMode {
 
 impl ClusterConnection {
     pub(crate) fn new(
-        initial_nodes: Vec<ConnectionInfo>,
-        read_from_replicas: bool,
-        username: Option<String>,
-        password: Option<String>,
+        cluster_params: &ClusterParams,
```

Comment:
I'm not sure there's a reason to use references for the params here, given that you're just cloning anyway. I would let the caller do the cloning if needed.

Reply:
I prefer to pass references as much as possible so that if I don't need a clone later on, I don't have to modify the whole call chain. Also if I only need a part of the struct, I don't have to clone the whole struct.
This is a matter of personal opinion. If you prefer, I can modify the code to pass cloned struct.